helix-stdx/src/rope.rs

```diff
@@ -75,4 +93,48 @@ impl<'a> RopeSliceExt<'a> for RopeSlice<'a> {
             .position(|ch| !ch.is_whitespace())
             .map(|pos| self.len_chars() - pos - 1)
     }
+
+    /// returns the char idx of `byte_idx`, if `byte_idx` is
+    /// a char boundary this function behaves the same as `byte_to_char`
+    fn byte_to_next_char(self, mut byte_idx: usize) -> usize {
+        let (chunk, chunk_byte_off, chunk_char_off, _) = self.chunk_at_byte(byte_idx);
+        byte_idx -= chunk_byte_off;
+        let is_char_boundary =
+            is_utf8_char_boundary(chunk.as_bytes().get(byte_idx).copied().unwrap_or(0));
+        chunk_char_off + byte_to_char_idx(chunk, byte_idx) + !is_char_boundary as usize
+    }
+}
+
+/// copiefd froom std
```

Comment:
```suggestion
// copied from std
```

typos but also I think this can be a regular comment rather than doc since it's only interesting if you're looking at this block (rather than in an lsp hover for example)